packages/kit/src/utils/escape.js

```diff
@@ -1,27 +1,42 @@
-// dict from https://github.com/yahoo/serialize-javascript/blob/183c18a776e4635a379fdc620f81771f219832bb/index.js#L25
-/** @type {Record<string, string>} */
-const escape_json_in_html_dict = {
-	'<': '\\u003C',
-	'>': '\\u003E',
-	'/': '\\u002F',
-	'\u2028': '\\u2028',
-	'\u2029': '\\u2029'
-};
-
-const escape_json_in_html_regex = new RegExp(
-	`[${Object.keys(escape_json_in_html_dict).join('')}]`,
-	'g'
-);
-
 /**
- * Escape a JSONValue that's going to be embedded in a `<script>` tag
- * @param {import('types').JSONValue} val
+ * Generates a raw HTML string containing a safe script element carrying JSON data and associated attributes.
+ *
+ * It escapes all the special characters needed to guarantee the element is unbroken, but care must
+ * be taken to ensure it is inserted in the document at an acceptable position for a script element,
+ * and that the resulting string isn't further modified.
+ *
+ * Attribute names must be type-checked so we don't need to escape them.
+ *
+ * **INTERPOLATING THIS STRING INTO AN UNRELATED SCRIPT ELEMENT IS NOT SAFE.**
```

Comment:
I'm not sure what this line means. This is outputting a `<script>` tag. Are you thinking someone might try to put this `<script>` tag inside another `<script>` tag? I'm not quite sure what that would mean or why anyone would try to do that

Reply:
Yeah, it was supposed to empathize the raw html string is not properly vetted for every context. Maybe I should reword it.

It's not that people would intentionally try to do that, but marking a string as "safe for html" in your head is risky: each html context has different and incompatible escaping rules, so care must always be taken to properly place these strings in the document, even if something was "escaped."